### PR TITLE
[Fix] Reenable the symbolication to allow debugging and update its implementation.

### PR DESCRIPTION
**Pull request by mandel-macaque**
This is WIP. We are reenabling the symbolication and updating its output to the new definition.

### CODE REVIEW

File: tools/mtouch/Application.cs [hunk 1]
```diff
@@ -1335,6 +1336,49 @@ static int GetIndentation (string line)
...
+							if (cmd is UuidCommand) {
+								var uuidCmd = cmd as UuidCommand;
+								root.Add (new XElement ("build-id", Encoding.UTF8.GetString (uuidCmd.uuid)));
+							}
+						}
+					}
+				}
+			}
+			// WriteAllText will replace the file if needed.
+			File.WriteAllText (manifestPath, root.ToString ());
```
Comment (rolfbjarne) [inline tools/mtouch/Application.cs]: Use `Driver.WriteIfDifferent` instead to write the contents, which doesn't write anything if the contents haven't changed.

File: tools/mtouch/mtouch.csproj [hunk 2]
```diff
@@ -344,16 +345,6 @@
     <Folder Include="Xamarin.Linker\" />
   </ItemGroup>
   <ItemGroup>
```
Comment (rolfbjarne) [inline tools/mtouch/mtouch.csproj]: You're removing unrelated chunks here.

Comment (mandel-macaque) [inline tools/mtouch/mtouch.csproj]: That was XS, I'll fix that.

**Comment by rolfbjarne** [comment]
There are msym tests: https://github.com/xamarin/xamarin-macios/blob/master/tests/mtouch/MTouch.cs#L132, please verify and update the tests as required as well.

**Comment by xamarin-release-manager** [comment]
Build success

File: tools/mtouch/Application.cs [hunk 3]
```diff
@@ -1335,6 +1336,49 @@ static int GetIndentation (string line)
 			};
 		}
 
+		// This method generates the manifes that is required by the symbolication in order to be able to debug the application,
```
Comment (spouliot) [inline tools/mtouch/Application.cs]: typo `manifes` (missing `t`)

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by spouliot** [comment]
Merging so QA can have a build asap for testing.
@mandel-macaque please do a separate PR for the tests that @rolfbjarne mentioned above. thanks!

**Pull request merged**